lib/acts_as_ar_query.rb

```diff
@@ -0,0 +1,205 @@
+require 'forwardable'
+# allow active record dsl to call legacy find
+class ActsAsArQuery
+  extend Forwardable
+  attr_accessor :klass, :mode, :options
+
+  # methods that execute actual query
+
+  # - [X] all
+  # - [X] count
+  # - [X] find
+  # - [ ] find_by
+  # - [X] first
+  # - [X] last
+  # - [X] size
+  # - [X] take
+
+  # methods that enhance / chain the query
+  # list is from ActiveRecord relation interface
+
+  # - [ ] bind
+  # - [ ] create_with
+  # - [ ] distinct
+  # - [ ] eager_load
+  # - [X] except ? - is this not defined in the interface?
+  # - [ ] extending
+  # - [ ] from
+  # - [X] group
+  # - [ ] ~~having~~ - NO
+  # - [X] includes (partial)
+  # - [ ] joins
+  # - [X] limit
+  # - [ ] lock
+  # - [.] none
+  # - [X] offset
+  # - [X] order (partial)
+  # - [ ] preload
+  # - [ ] readonly
+  # - [X] references (partial)
+  # - [X] reorder
+  # - [ ] reverse_order
+  # - [X] select (partial)
+  # - [X] unscope
+  # - [ ] uniq
+  # - [X] where (partial)
+  # - [ ] where.not
+
+  # private api
+  # the common pattern is to allow this method to be called frequently
+  # and add the values into the hash
+  # this is used by methods like includes, references, order, select and others
+  def append_hash_arg(*val)
+    symbol = __callee__
+    val = val.flatten
+    if val.first.kind_of?(Hash)
+      raise ArgumentError, "Need to support #{symbol}(#{val.class.name})"
+    end
+    dup.tap do |r|
+      r.options[symbol] = r.options[symbol] ? (r.options[symbol] + val) : val
+    end
+  end
+
+  def assign_arg(val)
+    dup.tap do |r|
+      r.options[__callee__] = val
+    end
+  end
```

Comment:
I would rather see explicit public calls to explicit private methods than caring about saving some lines by forcing aliases and thus publicizing a private interface.

That is, instead of aliases:

```ruby
def select
  append hash_arg
end

private 

def append hash arg; end
```